packager/mpd/base/mpd_builder.cc

```diff
@@ -75,57 +73,60 @@ bool Positive(double d) {
 
 // Return current time in XML DateTime format. The value is in UTC, so the
 // string ends with a 'Z'.
-std::string XmlDateTimeNowWithOffset(
-    int32_t offset_seconds,
-    base::Clock* clock) {
-  base::Time time = clock->Now();
-  time += base::TimeDelta::FromSeconds(offset_seconds);
-  base::Time::Exploded time_exploded;
-  time.UTCExplode(&time_exploded);
-
-  return base::StringPrintf("%4d-%02d-%02dT%02d:%02d:%02dZ", time_exploded.year,
-                            time_exploded.month, time_exploded.day_of_month,
-                            time_exploded.hour, time_exploded.minute,
-                            time_exploded.second);
+std::string XmlDateTimeNowWithOffset(int32_t offset_seconds, Clock* clock) {
+  auto time_t = std::chrono::system_clock::to_time_t(
+      clock->now() + std::chrono::seconds(offset_seconds));
+  std::tm* tm = std::gmtime(&time_t);
+
+  std::stringstream ss;
+  ss << std::put_time(tm, "%Y-%m-%dT%H:%M:%SZ");
+  return ss.str();
 }
 
 bool SetIfPositive(const char* attr_name, double value, XmlNode* mpd) {
   return !Positive(value) ||
          mpd->SetStringAttribute(attr_name, SecondsToXmlDuration(value));
 }
 
-std::string MakePathRelative(const std::string& media_path,
-                             const FilePath& parent_path) {
-  FilePath relative_path;
-  const FilePath child_path = FilePath::FromUTF8Unsafe(media_path);
-  const bool is_child =
-      parent_path.AppendRelativePath(child_path, &relative_path);
-  if (!is_child)
-    relative_path = child_path;
-  return relative_path.NormalizePathSeparatorsTo('/').AsUTF8Unsafe();
+bool IsBasePathParent(const std::filesystem::path& path,
+                      const std::filesystem::path& base_path) {
+  std::string path_str = path.string();
+  std::string base_path_str = base_path.string();
+  return path_str.substr(0, base_path_str.length()) == base_path_str;
+}
+
+std::string MakePathRelative(const std::filesystem::path& media_path,
```

Comment:
Right.  That's what the snippet I gave should do.

```c++
  // media_path = "foo/bar/baz/manifest.mpd", parent_path = "foo/bar"
  auto relative_path = std::filesystem::relative(media_path, parent_path);
  // relative_path = "baz/manifest.mpd"

  if (relative_path.empty() || *relative_path.begin() == "..") {
    // Not related.
    return media_path;
  }

  // returns "baz/manifest.mpd" here
  return relative_path;
```

If it seems like this method should only ever get inputs that follow the happy path, we could treat the other possibilities as failures instead:

```c++
  auto relative_path = std::filesystem::relative(media_path, parent_path);
  CHECK(!relative_path.empty());
  CHECK(*relative_path.begin() != "..");
  return relative_path;
```

Reply:
Ah, yes, I should have read more closely. I did try that approach as well of using `std::filesystem::relative` but I wasn't sure whether this would work in a cross-platform way. I can switch to this though if this should work on Windows as well.

Reply:
It should... I haven't tested any of the snippets I wrote.  But std::filesystem is supposed to be platform-aware, and many of the examples show APIs that would only do useful things on Windows and not Linux.  (Like splitting off drive letters from paths, etc).

Reply:
ok, updated to use this, we'll see if the windows tests pass